src/lib/hooks/useMessageTemplateUtils.ts

```diff
@@ -0,0 +1,176 @@
+  initializeMessageTemplatesInfo: (readySdk: SendbirdChat) => Promise<void>;
+}
+
+const {
+  INITIALIZE_MESSAGE_TEMPLATES_INFO,
+  UPSERT_MESSAGE_TEMPLATE,
+  UPSERT_WAITING_TEMPLATE_KEY,
+  MARK_ERROR_WAITING_TEMPLATE_KEY,
+} = APP_INFO_ACTIONS;
+
+export default function useMessageTemplateUtils({
+  sdk,
+  logger,
+  appInfoStore,
+  appInfoDispatcher,
+}): UseMessageTemplateUtilsWrapper {
+  const messageTemplatesInfo: MessageTemplatesInfo | undefined = appInfoStore?.messageTemplatesInfo;
+
+  const getCachedTemplate = (key: string): ProcessedMessageTemplate | null => {
+    if (!messageTemplatesInfo) return null;
+
+    let cachedTemplate: ProcessedMessageTemplate | null = null;
+    const cachedMessageTemplates: Record<string, ProcessedMessageTemplate> | null = messageTemplatesInfo?.templatesMap ?? null;
+    if (cachedMessageTemplates) {
+      cachedTemplate = cachedMessageTemplates[key] ?? null;
+    }
+    return cachedTemplate;
+  };
+
+  /**
+   * Fetches a single message template by given key and then
+   * returns processed template for updating templates info in global state.
+   * If no such templates exists (error) or any error occurs in response, return null.
+   */
+
+  const fetchAllMessageTemplates = async (readySdk: SendbirdChat): Promise<SendbirdMessageTemplate[]> => {
+    let hasMore = true;
+    let paginationToken = null;
+    const fetchedTemplates: SendbirdMessageTemplate[] = [];
+
+    while (hasMore) {
+      /**
+       * RFC doc:
+       * https://sendbird.atlassian.net/wiki/spaces/PLAT/pages/2254405651/RFC+Message+Template#%5BAPI%5D-List-message-templates
+       */
+      const res: MessageTemplateListResult = await readySdk!.message.getMessageTemplatesByToken(
+        paginationToken,
+        { limit: MESSAGE_TEMPLATES_FETCH_LIMIT },
+      );
+      hasMore = res.hasMore;
+      paginationToken = res.token;
+      res.templates.forEach((messageTemplate) => {
+        fetchedTemplates.push(JSON.parse(messageTemplate.template));
+      });
+    }
+    return fetchedTemplates;
+  };
+
+  const initializeMessageTemplatesInfo = async (readySdk: SendbirdChat): Promise<void> => {
+    const sdkMessageTemplateToken = readySdk!.appInfo?.messageTemplateInfo.token;
+
+    /**
+     * no sdkMessageTemplateToken => no templates => clear cached
+     */
+    if (!sdkMessageTemplateToken) {
+      localStorage.removeItem(CACHED_MESSAGE_TEMPLATES_TOKEN_KEY);
+      localStorage.removeItem(CACHED_MESSAGE_TEMPLATES_KEY);
+      return;
+    }
+    /**
+     * Given the following cases:
+     * 1. non-null sdkMessageTemplateToken => templates exist
+     * 2. no cached token or cached token is outdated => first fetch or outdated cache
+     *
+     * If both 1 and 2, fetch all templates and upsert to cache.
+     * If cached token is not outdated, use cached templates.
+     */
+    const cachedMessageTemplatesToken: string | null = localStorage.getItem(CACHED_MESSAGE_TEMPLATES_TOKEN_KEY);
```

Comment:
Since it was stored after being stringified with JSON.stringify, it seems necessary to parse it with JSON.parse before comparing it with the token retrieved from the SDK

Reply:
<img width="276" alt="image" src="https://github.com/sendbird/sendbird-uikit-react/assets/26326015/58d51e9a-7e4c-4c20-9d1b-585fd66808e5">

Reply:
token is string

Reply:
```
// In SDK
declare class MessageTemplateInfo {
  readonly token: string;
}
```

Reply:
When you stringify a string, it transforms as follows:
`"stringifiedString"` -> `"'stringifiedString'"`

Reply:
or you can edit line 112


```diff
- localStorage.setItem(CACHED_MESSAGE_TEMPLATES_TOKEN_KEY, JSON.stringify(sdkMessageTemplateToken));
+ localStorage.setItem(CACHED_MESSAGE_TEMPLATES_TOKEN_KEY, sdkMessageTemplateToken);
```